packages/venia-concept/src/components/PurchaseDetailsPage/OrderItem/orderItem.js

```diff
@@ -1,12 +1,103 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import { Price } from '@magento/peregrine';
+import classify from 'src/classify';
+import defaultClasses from './orderItem.css';
+import ButtonsGroup from './ButtonsGroup';
 
-export default class OrderItem extends Component {
+class OrderItem extends Component {
     static propTypes = {
-        item: PropTypes.shape({})
+        classes: PropTypes.shape({
+            root: PropTypes.string
+        }),
+        withBorders: PropTypes.bool,
+        item: PropTypes.shape({
+            title: PropTypes.string,
+            size: PropTypes.string,
+            color: PropTypes.string,
+            qty: PropTypes.number,
+            titleImageSrc: PropTypes.string,
+            price: PropTypes.number
+        }),
+        onBuyAgain: PropTypes.func,
+        onShare: PropTypes.func,
+        currencyCode: PropTypes.string
+    };
+
+    //TODO: get currencyCode whether from item object or from order or from user cart
+    static defaultProps = {
+        withBorders: false,
+        currencyCode: 'USD'
+    };
+    //TODO: make correct mapping from item
+    onBuyAgainHandler = () => {
+        this.props.onBuyAgain(this.props.item);
+    };
+    //TODO: make correct mapping from item
+    onShareHandler = () => {
+        this.props.onShare(this.props.item);
     };
 
     render() {
-        return <div>Order item</div>;
+        const {
+            classes,
+            item: { title, size, color, qty, titleImageSrc, price },
+            currencyCode,
+            withBorders
+        } = this.props;
+        const butonsGroupItems = [
```

Comment:
Please, move logic of generating items to helpers. So, add getItemButtons helper function, which will receive onBuyAgainHandler, onShareHandler, onReviewHandler, and return array of buttons.